ui/appui-react/src/appui-react/cursor/cursorpopup/CursorPopup.tsx

```diff
@@ -104,61 +105,112 @@ export class CursorPopup extends React.Component<
     pt: XAndY,
     offset: XAndY,
     popupSize: SizeProps | undefined,
-    relativePosition: RelativePosition
+    relativePosition: RelativePosition | Placement
   ): RectangleProps {
     const popupRect = { top: 0, left: 0, right: 0, bottom: 0 };
-
     if (popupSize === undefined) return popupRect;
 
-    switch (relativePosition) {
-      case RelativePosition.Top:
-        popupRect.bottom = pt.y - offset.y;
-        popupRect.left = pt.x - popupSize.width / 2;
-        popupRect.top = popupRect.bottom - popupSize.height;
-        popupRect.right = popupRect.left + popupSize.width;
-        break;
-      case RelativePosition.Left:
-        popupRect.right = pt.x - offset.x;
-        popupRect.top = pt.y - popupSize.height / 2;
-        popupRect.left = popupRect.right - popupSize.width;
-        popupRect.bottom = popupRect.top + popupSize.height;
-        break;
-      case RelativePosition.Right:
-        popupRect.left = pt.x + offset.x;
-        popupRect.top = pt.y - popupSize.height / 2;
-        popupRect.right = popupRect.left + popupSize.width;
-        popupRect.bottom = popupRect.top + popupSize.height;
-        break;
-      case RelativePosition.Bottom:
-        popupRect.top = pt.y + offset.y;
-        popupRect.left = pt.x - popupSize.width / 2;
-        popupRect.bottom = popupRect.top + popupSize.height;
-        popupRect.right = popupRect.left + popupSize.width;
-        break;
-      case RelativePosition.TopLeft:
-        popupRect.bottom = pt.y - offset.y;
-        popupRect.right = pt.x - offset.x;
-        popupRect.top = popupRect.bottom - popupSize.height;
-        popupRect.left = popupRect.right - popupSize.width;
-        break;
-      case RelativePosition.TopRight:
-        popupRect.bottom = pt.y - offset.y;
-        popupRect.left = pt.x + offset.x;
-        popupRect.top = popupRect.bottom - popupSize.height;
-        popupRect.right = popupRect.left + popupSize.width;
-        break;
-      case RelativePosition.BottomLeft:
-        popupRect.top = pt.y + offset.y;
-        popupRect.right = pt.x - offset.x;
-        popupRect.bottom = popupRect.top + popupSize.height;
-        popupRect.left = popupRect.right - popupSize.width;
-        break;
-      case RelativePosition.BottomRight:
-        popupRect.top = pt.y + offset.y;
-        popupRect.left = pt.x + offset.x;
-        popupRect.bottom = popupRect.top + popupSize.height;
-        popupRect.right = popupRect.left + popupSize.width;
-        break;
+    if (typeof relativePosition === "string") {
+      const placement = relativePosition.split("-");
+      const [mainPlacement, subPlacement] = placement;
+
+      switch (mainPlacement) {
```

Comment:
nit: error prone duplication, instead we could allways map: `relativePosition` to `placement | X` (where X is a relative position that can't be mapped to placement). And just handle `placement` values and a few of RelativePosition values.

Reply:
Oh, I assumed there is an overlap, if that's not the case - feel free to ignore :)